# Evaluation of hourly AUC (Area Under Curve) results

From ChatGPT:

| Glycemic State     | Avg Glucose (mg/dL) | AUC (mg·min/dL per hour) |
| ------------------ | ------------------- | ------------------------ |
| Normal             | 80–100              | 4800–6000                |
| Slightly elevated  | 100–140             | 6000–8400                |
| Postprandial spike | 140–180             | 8400–10,800              |
| Hyperglycemia      | >180                | >10,800                  |


In [1]:
%reload_ext autoreload
%autoreload 2

test_data = "../tests/data/example_data_5_subject.csv"

# ChatGPT calculations  

**Average Hourly AUC Per Subject (for test_data)**

| Subject       | Average Hourly AUC | Interpretation                                                                                             |
| ------------- | ------------------ | ---------------------------------------------------------------------------------------------------------- |
| **Subject 1** | 6,241              | ✅ Near-normal range — mild elevation. Likely good glucose control.                                         |
| **Subject 2** | 11,825             | 🔴 High AUC — consistent with **hyperglycemia**, possibly uncontrolled diabetes or postprandial spikes.    |
| **Subject 3** | 8,043              | 🟠 Moderately elevated — suggests elevated average glucose, likely post-meal spikes or insulin resistance. |
| **Subject 4** | 7,041              | 🟡 Slightly elevated — above normal but not extreme. Possibly impaired glucose tolerance.                  |
| **Subject 5** | 9,302              | 🔴 High — consistent with sustained hyperglycemia; could indicate diabetes or poor control.                |






# IGLU/IGLU-PY results

In [ ]:
import sys
from importlib.metadata import version

import pandas as pd
import rpy2.robjects as ro
import iglu_py 


In [3]:
# Print versions for future references
print(f"Python version: {sys.version}")
print(f"R version: {ro.r('R.version.string')}")
iglu_version = str(ro.r('packageVersion("iglu")'))
print(f"iglu version: {iglu_version}")
print(f"iglu_py version: {version('iglu-py')}")
print(f"rpy2 version: {version('rpy2')}")

Python version: 3.11.10 (main, Oct  3 2024, 02:26:51) [Clang 14.0.6 ]
R version: [1] "R version 4.4.3 (2025-02-28)"

iglu version: [1] ‘4.2.2’

iglu_py version: 1.1.1
rpy2 version: 3.6.0


In [4]:
# load test data into DF
df = pd.read_csv(test_data, index_col=0)

auc_results = iglu_py.auc(df)
auc_results =auc_results.round(0)
display(auc_results)



id  hourly_auc
1  Subject 1       123.0
2  Subject 2       218.0
3  Subject 3       154.0
4  Subject 4       130.0
5  Subject 5       174.0

## Conclusions 
IGLU AUC calculations are substantially differ from expected ranges suggested by ChatGPT

| Subject       | IGLU  | ChatGPT  |
|:------------- | -----:| --------:|
| Subject 1	    | 123   | 6,241    |
| Subject 2	    | 218   | 11,825   |
| Subject 3	    | 154   | 8,043    |
| Subject 4	    | 130   | 7,041    |           
| Subject 5	    | 174   | 9,302    |

# IGLU_PYTHON results

In [5]:
# Add project directory to PYTHONPATH
import sys
import os
sys.path.append(os.path.abspath('..'))

In [6]:
import pandas as pd
import iglu_python

# load test data into DF
df = pd.read_csv(test_data, index_col=0)

auc_results = iglu_python.auc(df)
auc_results = auc_results.round(0)
display(auc_results)





id  hourly_auc
0  Subject 1      7002.0
1  Subject 2     12902.0
2  Subject 3      8829.0
3  Subject 4      7751.0
4  Subject 5     10242.0

## Conclusions  
IGLU_PYTHON AUC calculations are substantially differ from IGLU calculations, but close to  ranges suggested by ChatGPT


| Subject       | IGLU  | IGLU_PYTHON | ChatGPT  |
|:------------- | -----:| -----------:| --------:|
| Subject 1     |   123 |      7,002  |    6,241 |
| Subject 2     |   218 |     12,902  |   11,825 |
| Subject 3     |   154 |      8,829  |    8,043 |
| Subject 4     |   130 |      7,751  |    7,041 |           
| Subject 5     |   174 |     10,242  |    9,302 |
